core/src/datayoga_core/utils.py

```diff
@@ -79,11 +79,16 @@ def get_resource_path(relative_path: str) -> str:
         return path.join(os.path.dirname(__file__), "resources", relative_path)
 
 
-def split_field(field: str) -> List[str]:
-    return re.split(r"(?<!\\)\.", field)
+def split_field(field: str, *, __expression=re.compile(r"(?<!\\)\.")) -> List[str]:
```

Comment:
It's the way to avoid using an external variable to store the expression. The default value acts like cache and double underscore indicate that this arg must not be used by caller. Compiled expressions working faster.